scripts/which_r.sh

```diff
@@ -1,66 +1,102 @@
+  echo "  1. R $built_in_r_version (release version built into this container)"
+
+  # Check for additional R versions in subdirectories
+  if [ -d "$parent_dir" ]; then
+    # Create an array to store subdirectory names
+    subdirs=()
+
+    # Loop through subdirectories and print numbered list
+    counter=2  # Start counter at 2 to avoid conflict with built-in R
+    for dir in "$parent_dir"/*; do
+      if [ -d "$dir/bin" ] && [ -x "$dir/bin/R" ]; then
+        subdir=$(basename "$dir")
+        subdirs+=("$subdir")  # Populate subdirs array
+        echo "  $counter. $subdir"
+        ((counter++))
+      fi
+    done
+  fi
+
+  # If no additional R builds were found
+  if [ ${#subdirs[@]} -eq 0 ]; then
+    range=1
+    echo "No additional R builds available."
   else
-    echo "Invalid choice, please select options between 1 to $range. Defaulting to built-in R version."
+    range=$((counter - 1))
   fi
-  selected_version="/usr/bin/R"
-fi
 
-# Update settings.json with the chosen R path
-updated_settings_data=$(cat "$settings_file_path" | jq --arg subdir "$selected_version" '."r.rterm.linux"=$subdir')
-echo "$updated_settings_data" > "$settings_file_path"
+  # Get user choice
+  read -p "Enter the number corresponding to the selected version: " choice
 
-echo "R terminal will now use version: $selected_version"
+  # Define selected version based on choice
+  if [[ "$choice" -eq 1 ]]; then
+    # Use built-in R
+    selected_version="/usr/bin/R"
+  elif [[ "$choice" -ge 2 ]] && [[ "$choice" -lt "$counter" ]]; then
+    # Use R from chosen subdirectory
+    chosen_subdir="${subdirs[((choice - 2))]}"
+    selected_version="$parent_dir/$chosen_subdir/bin/R"
+  else
+    # Invalid choice, default to built-in R
+    if [[ $range -eq 1 ]]; then
+      echo "Invalid choice, please enter 1. Defaulting to built-in R version."
+    else
+      echo "Invalid choice, please select options between 1 to $range. Defaulting to built-in R version."
+    fi
+    selected_version="/usr/bin/R"
+  fi
+
+  # Update settings.json with the chosen R path
+  updated_settings_data=$(cat "$settings_file_path" | jq --arg subdir "$selected_version" '."r.rterm.linux"=$subdir')
+  echo "$updated_settings_data" > "$settings_file_path"
+
+  echo "R terminal will now use version: $selected_version"
+}
+
+which_r() {
+  if [ "$HOME" = '/home/gitpod' ]; then
+    NEWDIR="$WORK_DIR/.vscode"
+    NEWFILE="$NEWDIR/settings.json"
+
+    # Create the new directory
+    mkdir -p $NEWDIR
+
+    # Write text directly into the new file using cat and a heredoc
+    cat <<EOL > $NEWFILE
+{
+  "r.lsp.diagnostics": false,
+  "r.plot.useHttpgd": true,
+  "r.rpath.linux": "/usr/bin/R",
+  "r.rterm.linux": "/usr/bin/R",
+  "terminal.integrated.sendKeybindingsToShell": true,
+  "svn.multipleFolders.enabled": true,
+  "workbench.editorAssociations": {
+    "*.md": "vscode.markdown.preview.editor"
+  },
+  "workbench.welcomePage.walkthroughs.openOnInstall": false
+}
+EOL
```

Comment:
This code means that you are creating the `.vscode` directory and the `settings.json` file are only created when `which_r()` is run.

Since we want the settings to be there from when the codespace is first created, I think you should create an initial version of this file as part of `localscript.sh`. Then the code in `which_r.sh` could be much simplified as it is only the file path that changes between GitHub and GitPod, in particular, you may be able to simplify this code to one function, rather than needing the separate `set_r()` function.

To avoid maintaining two copies of the VS Code settings - one in `.devcontainer/devcontainer.json` and one in `.vscode/settings.json`, maybe you can create the initial version of `.vscode/settings.json` from the settings in `.devcontainer/devcontainer.json`using jq.

I think having an initial version of `.vscode/settings.json` on startup may fix the issue with the markdown preview not working when INDEX.md is opened at the start.